dlt/common/cli/runner/runner.py

```diff
@@ -0,0 +1,32 @@
+import typing as t
+
+from dlt.common.cli.runner.inquirer import Inquirer
+from dlt.common.pipeline import LoadInfo
+from dlt.common.cli.runner.pipeline_script import PipelineScript
+from dlt.common.cli.runner.types import RunnerParams
+from dlt.extract.resource import DltResource
+
+
+class PipelineRunner:
+    def __init__(self, params: RunnerParams) -> None:
+        self.params = params
+        self.script = PipelineScript(params)
+        self.params.pipeline_workdir = self.script.workdir
+        self.inquirer = Inquirer(self.params, self.script.pipeline_members)
+
+    def __enter__(self) -> t.Self:
+        self.inquirer.preflight_checks()
+        self.inquirer.check_if_runnable()
+        self.pipeline, self.resource = self.inquirer.maybe_ask()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
+
+    def run(self) -> LoadInfo:
+        data_resource = self.resource
+        if isinstance(self.resource, DltResource) and not self.resource._args_bound:
+            data_resource = self.resource()
```

Comment:
@sh-rp @rudolfix how to check if resource generator opened state for example
```py
@dlt.resource()
def people_names():
    yield ...

booba_names = people_names()
```
while for the people_names we can just run it like `pipeline.run(people_names())` for the latter we just need to pass it as is `pipeline.run(booba_names)`
if I check if `resource._args_bound == True` will it be sufficient?